Packs/CofenseTriage/Integrations/CofenseTriage/CofenseTriage.py

```diff
@@ -51,145 +25,248 @@
     'category_id',
     'match_priority',
     'tags',
-    'email_attachments'
+    'email_attachments',
 ]
 
 
-# HELPER FUNCTIONS #
-def snake_to_camel_keys(snake_list: List[Dict]) -> List[Dict]:
-    def snake_to_camel(snake_str) -> str:
-        if snake_str == 'id':
-            return 'ID'
-        components = snake_str.split('_')
-        return ''.join(x.title() for x in components)
-
-    return [{snake_to_camel(k): v for k, v in snake_d.items()} for snake_d in snake_list]
-
-
-def split_snake(string: str) -> str:
-    return string.replace("_", " ").title()
+class TriageInstance:
+    def __init__(self, *, host, token, user, disable_tls_verification=False):
+        self.host = host
+        self.token = token
+        self.user = user
+        self.disable_tls_verification = disable_tls_verification
 
+    def request(self, endpoint, params=None, body=None, raw_response=False):
+        """
+        Make a request to the configured Triage instance and return the result.
+        """
+        # TODO automatic rate-limiting
+        response = requests.get(
+            self.api_url(endpoint),
+            headers={
+                "Authorization": f"Token token={self.user}:{self.token}",
+                "Accept": "application/json",
+            },
+            params=params,
+            data=body,
+            verify=not self.disable_tls_verification,
+        )
 
-# MAIN FUNCTIONS #
-def http_request(url_suffix: str, params=None, body=None, raw_response=False) -> Any:
-    """
-    Generic request to Cofense Triage. Client applications can make 25 requests to Cofense Triage
-    within a five-minute interval using the Cofense Triage API.
-    """
-    response = requests.get(
-        BASE_URL + url_suffix,
-        headers=HEADERS,
-        params=params,
-        data=body,
-        verify=USE_SSL,
-    )
-    try:
         if not response.ok:
-            return_error(f'Call to Cofense Triage failed [{response.status_code}] - [{response.text}]')
+            return return_error(
+                f"Call to Cofense Triage failed ({response.status_code}): {response.text}"
+            )
```

Comment:
please use `raise`, and create a main for the command_name if/else block.
we will try/catch the raise there.

Reply:
I'm not sure I understand the suggestion. Something like this (untested)?

```diff
diff --git a/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py b/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py
index dee2640d5..8c705af31 100644
--- a/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py
+++ b/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py
@@ -29,6 +29,15 @@ TERSE_FIELDS = [
 ]


+class TriageRequestFailedError(Exception):
+    def __init__(self, status_code, message):
+        self.status_code = status_code
+        self.message = message
+
+    def __str__(self):
+        return f"Call to Cofense Triage failed ({self.status_code}): {self.message}"
+
+
 class TriageInstance:
     def __init__(self, *, host, token, user, disable_tls_verification=False):
         self.host = host
@@ -53,9 +62,7 @@ class TriageInstance:
         )

         if not response.ok:
-            return return_error(
-                f"Call to Cofense Triage failed ({response.status_code}): {response.text}"
-            )
+            raise TriageRequestFailedError(response.status_code, response.text)

         if response.status_code == 206:
             # 206 indicates Partial Content. The reason will be in the warning header.
@@ -186,6 +193,15 @@ TRIAGE_INSTANCE = TriageInstance(
 )


+def triage_instance_request_wrapper(*args, **kwargs):
+    try:
+        response = TRIAGE_INSTANCE.request(*args, **kwargs)
+    except TriageRequestFailedError as e:
+        return return_error(e.message)
+
+    return  response
+
+
```

And then every invocation would be replaced by this?

```diff
diff --git a/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py b/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py
index dee2640d5..25e4b92dc 100644
--- a/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py
+++ b/Packs/CofenseTriage/Integrations/CofenseTriage2/CofenseTriage.py
@@ -208,7 +224,7 @@ def parse_triage_date(date: str):

 def test_function() -> None:
     try:
-        response = TRIAGE_INSTANCE.request("processed_reports")
+        response = triage_instance_request_wrapper("processed_reports")
```

Is there an advantage to this additional layer of indirection?

Reply:
the try/catch should be in the buttom of the file code, in the main.
Yes, it will make the errors printed to the user in the war room more readable, as it will not include the stacktrace.